speechbrain/lobes/models/huggingface/transformers.py

```diff
@@ -0,0 +1,274 @@
+                cache_dir=cache_dir,
+            )
+
+        # Adjust model as desired
+        if override_hf_model_partial_fn is not None:
+            self.model = override_hf_model_partial_fn(self.model)
+
+        # Set inner forward function
+        if forward_partial_fn is not None:
+            if isinstance(forward_partial_fn, partial):
+                self.forward_partial_fn = forward_partial_fn
+                self.forward_partial_fn.keywords["model"] = self.model
+            else:
+                self.forward_partial_fn = partial(
+                    forward_partial_fn, model=self.model
+                )
+        else:
+            self.forward_partial_fn = partial(default_forward, model=self.model)
+
+        # Prepare for training, fine-tuning, or inference
+        self.freeze = freeze
+        if self.freeze:
+            logger.warning(
+                "speechbrain.lobes.models.HuggingFaceTransformer is frozen."
+            )
+            self.model.eval()
+            for param in self.model.parameters():
+                param.requires_grad = False
+        else:
+            self.model.gradient_checkpointing_disable()  # Required by DDP
+            self.model.train()
+            if freeze_nested_models_their_calls is not None:
+                if type(freeze_nested_models_their_calls) is not list:
+                    freeze_nested_models_their_calls = [
+                        freeze_nested_models_their_calls
+                    ]
+                for nested_freeze_call in freeze_nested_models_their_calls:
+                    eval(f"self.model.{nested_freeze_call}()")
+
+    def _from_pretrained(
+        self,
+        source,
+        config,
+        model,
+        modify_state_dict_partial_fn,
+        save_path,
+        cache_dir,
+    ):
+        """This function manages the source checking and loading of the params.
+        # 1. Is the model from HF or a local path
+        # 2. Is the model pretrained with HF or SpeechBrain
+        # 3. Download (if appropriate) and load with respect to 1. and 2.
+        """
+        is_sb, ckpt_file = _check_model_source(source, save_path)
+        if is_sb:
+            config = config.from_pretrained(source, cache_dir=save_path)
+            self.model = model(config)
+            self.model.gradient_checkpointing_disable()  # Required by DDP
+            # fetch the checkpoint file
+            ckpt_full_path = fetch(
+                filename=ckpt_file,
+                source=source,
+                savedir=save_path,
+                cache_dir=cache_dir,
+            )
+            # We transfer the parameters from the checkpoint.
+            self._load_sb_pretrained_parameters(
+                ckpt_full_path,
+                modify_state_dict_partial_fn=modify_state_dict_partial_fn,
+            )
+        else:
+            self.model = model.from_pretrained(source, cache_dir=save_path)
+
+    def _load_sb_pretrained_parameters(
```

Comment:
Well this was something that I created specifically for wav2vec2 pertaining. I have NO IDEA if it will work for others. xD

Reply:
either it is harmonised for all the interface interactions;
or put to another pretrained file;
or we put some handle here & put this function declaration to another lib_-style file.

wav2vec2 pretrainer - should it be handled through the wav2vec2 transformer interface - through another one - or completely different - what do you want here?

Reply:
How about making the entire `_from_pretrained` to another partial and the if/else construct could be two different "default" functions that are placed in the `overrides.py`?

Would prefer to have a generic interface with some "default" specifications/overrides, so that users have full flexibility here to use HF as they need and as HF progresses through their next versions.